test/hotspot/gtest/opto/test_regmask.cpp

```diff
@@ -170,28 +186,1037 @@ TEST_VM(RegMask, is_bound_pair) {
+        OptoReg::dump(reg);
+        tty->cr();
+      }
+      mask.Set_All_From(reg);
+      mask_ref.set_range(reg, offset_ref + size_bits_before);
+      infinite_stack_ref = true;
+      break;
+    default:
+      FAIL() << "Unimplemented action";
+    }
+    ASSERT_NO_FATAL_FAILURE(assert_equivalent(mask, mask_ref, infinite_stack_ref));
+  }
+}
+
+// Randomly sets register mask contents. Does not change register mask size.
+static void randomize(RegMask& rm) {
+  rm.Clear();
+  // Uniform distribution over number of registers.
+  uint regs = next_random() % (rm.rm_size_in_bits() + 1);
+  for (uint i = 0; i < regs; i++) {
+    uint reg = (next_random() % rm.rm_size_in_bits()) + rm.offset_bits();
+    rm.Insert(reg);
+  }
+  rm.set_infinite_stack(next_random() % 2);
+}
+
+static uint grow_randomly(RegMask& rm, uint min_growth = 1,
+                          uint max_growth = 3) {
+  // Grow between min_growth and max_growth times.
+  uint grow = min_growth + (max_growth > 0 ? next_random() % max_growth : 0);
+  for (uint i = 0; i < grow; ++i) {
+    uint reg = rm.rm_size_in_bits();
+    if (reg >= RegMask::gtest_rm_size_in_bits_max()) {
+      // Cannot grow more
+      break;
+    }
+    // Force grow
+    rm.Insert(reg);
+    if (!rm.is_infinite_stack()) {
+      // Restore
+      rm.Remove(reg);
+    }
+  }
+  // Return how many times we grew
+  return grow;
+}
+
+TEST_VM(RegMask, random_copy) {
+  init_random();
+
+  auto print_failure = [&](const RegMask& src, const RegMask& dst) {
+    tty->print_cr("Failure, src and dst not equal");
+    tty->print("src: ");
+    src.dump_hex();
+    tty->cr();
+    tty->print("dst: ");
+    dst.dump_hex();
+    tty->cr();
+  };
+
+  // Test copying a larger register mask
+  for (uint i = 0; i < iterations; i++) {
+    ResourceMark rm;
+
+    // Create source RegMask
+    RegMask src(arena());
+
+    // Grow source randomly
+    grow_randomly(src);
+
+    // Randomly initialize source
+    randomize(src);
+
+    // Copy construct source to destination
+    RegMask dst(src, arena());
+
+    // Check equality
+    bool passed = src.gtest_equals(dst);
+    if (Verbose && !passed) {
+      print_failure(src, dst);
+    }
+    ASSERT_TRUE(passed);
+  }
+
+  // Test copying a smaller register mask
+  for (uint i = 0; i < iterations; i++) {
+    ResourceMark rm;
+
+    // Create destination RegMask
+    RegMask dst(arena());
+
+    // Grow destination arbitrarily (1-3 times)
+    uint growth = grow_randomly(dst, 1, 3);
+
+    // Create source RegMask
+    RegMask src(arena());
+
+    // Grow source arbitrarily, but not as much as destination
+    grow_randomly(src, 0, growth - 1);
+
+    // Randomly initialize source
+    randomize(src);
+
+    // Copy source to destination
+    dst = src;
+
+    // Check equality
+    bool passed = src.gtest_equals(dst);
+    if (Verbose && !passed) {
+      print_failure(src, dst);
+    }
+    ASSERT_TRUE(passed);
+  }
+}
+
+#endif // !PRODUCT
```

Comment:
Optional:
You could add some tests that expect a vm assert. You can do that with `TEST_VM_ASSERT_MSG`. Example:
https://github.com/openjdk/jdk/blob/9b04b5a74cc09b64098fb9940aa224f529ff1a01/test/hotspot/gtest/utilities/test_growableArray.cpp

Reply:
Thanks, added a few obvious such tests!